src/main/java/fi/aalto/cs/apluscourses/intellij/activities/InitializationActivity.java

```diff
@@ -61,6 +64,13 @@ public void runActivity(@NotNull Project project) {
     PluginSettings.getInstance().createUpdatingMainViewModel(project);
     ActionUtil.launch(RequiredPluginsCheckerAction.ACTION_ID,
         new ExtendedDataContext().withProject(project));
+
+    // todo: this validation should be somewhere in the Project (A+) class
```

Comment:
still thinking where to drop such a check... should a project (A+ project e.g.) know if the scope has been turned into an "A+ project" already... ?

Reply:
Isn't the current solution quite good, since the existence of a course file is a good way to check if it's a course project or not?